gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -94,22 +98,32 @@ private List<String> getFieldNames(Field f) {
     return fieldNames;
   }
 
-  @Override public <T> TypeAdapter<T> create(Gson gson, final TypeToken<T> type) {
+  @Override
+  public <T> TypeAdapter<T> create(Gson gson, final TypeToken<T> type) {
     Class<? super T> raw = type.getRawType();
 
     if (!Object.class.isAssignableFrom(raw)) {
       return null; // it's a primitive!
     }
 
-    FilterResult filterResult = ReflectionAccessFilterHelper.getFilterResult(reflectionFilters, raw);
+    if (ReflectionHelper.isRecord(raw)) {
+      // We can assume blockInaccessible here, because it will never kick inn. We are not doing
+      // direct field access, and are instead constructing records via their canonical constructor.
```

Comment:
> We are not doing direct field access

This seems to be wrong; for deserialization this implementation currently obtains the component values from the `private` fields.
Maybe it would be worth considering obtaining them from the accessor methods instead?

Reply:
Fixed